### PR TITLE
Breaking/nex 91/integrate npm item runner

### PR DESCRIPTION
**Pull request by oatymart**
Jira ticket: https://oat-sa.atlassian.net/browse/NEX-91
Vendorisation of item runner (reintegration)

**What I did:**
- deleted all js files & test files which were moved to `tao-item-runner-qti-fe`
- registered new npm paths in php via install script & updater
- registered new npm paths in `paths.json` for grunt test task (extractable by tao-core helper)
- created `package.json` to install the new npm package
- enabled composer-npm-bridge to automatically install packages

**Prerequisites:**
- [x] https://github.com/oat-sa/tao-core/pull/2195
- [x] https://github.com/oat-sa/tao-item-runner-fe/pull/6
- [x] https://github.com/oat-sa/tao-item-runner-qti-fe/pull/3
- [x] release & publish on npmjs `tao-item-runner` @ 0.3.0
- [x] release & publish on npmjs `tao-item-runner-qti` @ 0.3.0
- [x] update above references in this PR's package.json, & commit package-lock
- [x] also check the dependency to `tao` and `taoItems` in manifest.php, it may need updating...

**How to test:**
Install this version of the taoQtiItem extension in your TAO platform.
Verify that the `@oat-sa/tao-item-runner-qti` and `@oat-sa/tao-item-runner` npm packages were installed to views/node_modules.
Verify that TAO backoffice and test delivery still work with no 404 errors (DEBUG mode).
Verify that `npx grunt taoqtiitembundle` bundle generation task completes, and site works (PROD mode)
Verify that `npx grunt connect:dev taoqtiitemtest` finds and runs the remaining tests (I believe it's still necessary to create a `views/js/mathjax/MathJax.js` file for this to work)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -7,3 +7,4 @@ composer.lock
 /views/scss/qti.css.map
 /views/scss/item-creator.css
 /views/scss/item-creator.css.map
+/views/node_modules
```
Comment (btamas) [inline .gitignore]: I think you should ignore all `node_modules`.

File: composer.json [hunk 2]
```diff
@@ -52,13 +51,17 @@
 	"GPL-2.0-only"
 	],
     "extra" : {
-    	"tao-extension-name" : "taoQtiItem"
+			"tao-extension-name" : "taoQtiItem",
```
Comment (btamas) [inline composer.json]: The indentation looks really chaotic.

File: views/package.json [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  "license": "GPL-2.0",
+  "description": "taoQtiItem frontend modules",
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/oat-sa/extension-tao-itemqti.git"
+  },
+  "dependencies": {
+    "@oat-sa/tao-item-runner-qti": "0.3.0",
+    "@oat-sa/tao-item-runner": "0.3.0",
+    "@babel/polyfill": "^7.4.4"
```
Comment (btamas) [inline views/package.json]: Why do you need `@babel/polyfill` here?

Comment (krampstudio) [reply views/package.json]: not needed

File: scripts/install/RegisterNpmPaths.php [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+/**
+ * Register the paths of frontend modules which are installed from npm packages
+ *
+ * @author Martin Nicholson <martin@taotesting.com>
+ */
+class RegisterNpmPaths extends InstallAction
+{
+    /**
+     * @param $params
```
Comment (julien-sebire) [inline scripts/install/RegisterNpmPaths.php]: `$params` is never used but needed for the interface compliance. Do we need the docblock?

File: scripts/install/RegisterNpmPaths.php [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    {
+        $assetService = $this->getServiceManager()->get(AssetService::SERVICE_ID);
+        $taoQtiItemNpmDist = $assetService->getJsBaseWww('taoQtiItem') . 'node_modules/@oat-sa/tao-item-runner-qti/dist/';
+        $clientLibRegistry = ClientLibRegistry::getRegistry();
+        $clientLibRegistry->register('taoQtiItem/qtiCommonRenderer', $taoQtiItemNpmDist . 'qtiCommonRenderer');
+        $clientLibRegistry->register('taoQtiItem/qtiItem', $taoQtiItemNpmDist . 'qtiItem');
+        $clientLibRegistry->register('taoQtiItem/qtiRunner', $taoQtiItemNpmDist . 'qtiRunner');
+        $clientLibRegistry->register('taoQtiItem/runner', $taoQtiItemNpmDist . 'runner');
+        $clientLibRegistry->register('taoQtiItem/scoring', $taoQtiItemNpmDist . 'scoring');
+        return \common_report_Report::createSuccess('extra paths for taoQtiItem set up.');
```
Comment (julien-sebire) [inline scripts/install/RegisterNpmPaths.php]: Code style: please insert a blank line before `return` statement.

File: scripts/install/RegisterNpmPaths.php (hunk 4 above)
Comment (ferenckiss89) [reply scripts/install/RegisterNpmPaths.php]: fixed

File: scripts/install/RegisterNpmPaths.php (hunk 5 above)
Comment (ferenckiss89) [reply scripts/install/RegisterNpmPaths.php]: fixed

**Pull request merged**